tachyon/math/polynomials/univariate/univariate_evaluation_domain.h

```diff
@@ -224,17 +220,24 @@ class UnivariateEvaluationDomain : public EvaluationDomain<F, MaxDegree> {
       //    = (Z_H(𝜏) * gᵢ * v₀)⁻¹
       F l_i = (z_h_at_tau * omega_i).Inverse() * t;
       F negative_omega_i = -omega_i;
-      std::vector<F> lagrange_coefficients_inverse =
-          base::CreateVector(size, [this, &l_i, &tau, &negative_omega_i]() {
-            // 𝜏 - h * gⁱ
-            F r_i = tau + negative_omega_i;
-            // (Z_H(𝜏) * vᵢ)⁻¹ * (𝜏 - h * gⁱ)
-            F ret = l_i * r_i;
-            // lᵢ₊₁ = g⁻¹ * lᵢ
-            l_i *= group_gen_inv_;
-            // -h * gⁱ⁺¹
-            negative_omega_i *= group_gen_;
-            return ret;
+      std::vector<F> lagrange_coefficients_inverse(size);
+      base::Parallelize(
+          lagrange_coefficients_inverse,
+          [this, &l_i, &tau, &negative_omega_i](
+              absl::Span<F> chunk, size_t chunk_idx, size_t chunk_size) {
+            size_t n = chunk_idx * chunk_size;
+            // lᵢ * g⁻ⁱ
+            F l_i_pow = l_i * group_gen_inv_.Pow(n);
+            // - h * gⁱ
+            F negative_omega_i_pow = negative_omega_i * group_gen_.Pow(n);
+            for (F& c : chunk) {
+              // (Z_H(𝜏) * vᵢ)⁻¹ * (𝜏 - h * gⁱ)
+              c = l_i_pow * (tau + negative_omega_i_pow);
+              // lᵢ₊₁ = g⁻¹ * lᵢ
+              l_i_pow *= group_gen_inv_;
+              // - h * gⁱ⁺¹
```

Comment:
```suggestion
              // hᵢ₊₁ = -hᵢ * g⁻¹

Reply:
idk if this is right -> correct me if I'm wrong.

Reply:
It's actually wrong since `h` is the coset offset, which is a constant value, so hᵢ and hᵢ₊₁ technically doesn't make sense.

What I wanted to express was, from i-th coefficient, at this code, multiplying g, the result will be - h * gⁱ⁺¹.